### DISCUSSION TITLE
failed to initialize LSP for `source.toml`

### DISCUSSION THREAD

**Discussion by mkastner** [helix-editor/helix#4927] [Q&A] [answered]
I am somewhat befuddled.

I've removed the `~/.config/helix/languages.toml` to track some config error.

If I understand the documentation correctly, helix should use these internal language settings

https://github.com/helix-editor/helix/blob/master/languages.toml

as default if `~/.config/helix/languages.toml` does not exist.

Why do I get

```
2022-11-28T20:05:28.612 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
```

in the log file then?

**Comment by kirawi** [comment, marked as answer]
It is an error that will be reported if you don't have the LSP for the language installed. If you don't care about an LSP, then you can safely ignore t.

**Comment by mkastner** [comment]
@kirawi I was under the, obviously false, impression, that the languages which are listed here 

https://github.com/helix-editor/helix/blob/master/languages.toml

and which provide lsp support, already come with working language server connections.

Then, that means I still have to install the lsp-connectors and setups for each of these languages. Is that right?

**Reply by kirawi** [reply]
You just have to install the LSP servers. We have a wiki page for that: https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers

**Comment by mkastner** [comment]
@kirawi 
I was somehow under the impression, that helix would provide these installations. Gee, that's like (N)Vim's Coc with all its setups for each language all over again.

**Reply by kirawi** [reply]
Automatic installation will have to wait until a plugin system is implemented: https://github.com/helix-editor/helix/issues/2152
